src/plugins/selected-suggestion-ui.ts

```diff
@@ -132,10 +132,7 @@ export default (props: SelectedSuggestionProps) =>
           return buildSelectedSuggestionDecoration(suggestionId, state, props)
         }
 
-        if (
-          tr.getMeta(CLEAR_SUGGESTION_ID) ||
-          tr.getMeta('track-changes-refresh-changes')
-        ) {
```

Comment:
@asouqi do you know what the purpose of this check is? I removed it and the plugin seems to behave correctly. We should have a discussion about why we needed a plugin instead of pure HTML to achieve this functionality.

Reply:
this is to clear the selection when we change the state of the changes, so if you select a text and then approve it we should remove that selection.

> We should have a discussion about why we needed a plugin instead of pure HTML to achieve this functionality.

the plugin will give us out of the box click listener on the editor body that will help us to decide when the selection should fade away, also we are using plugin view which gives us a reach to the editor DOM. 

the plugin for this case is the best abstraction for the track change selection, is there a performance issue with that plugin?

Reply:
The check I removed specifically is causing performance issues, it triggers on every keypress resulting in a very sluggish typing experience.

Reply:
ok, will do that cleanup from the inspector. can please add this in article-editor
```
const tr = view.state.tr.setSelection(Selection.near(view.state.doc.resolve(view.state.selection.anchor)))
tr.setMeta("CLEAR_SUGGESTION_ID", true)
dispatch(tr);
```
https://github.com/Atypon-OpenSource/manuscripts-article-editor/blob/master/src/components/track-changes/TrackChangesPanel.tsx#L48-L52

Reply:
Done.